src/vs/platform/list/browser/listService.ts

```diff
@@ -40,6 +40,16 @@ export interface IListService {
 	 * Returns the currently focused list widget if any.
 	 */
 	readonly lastFocusedList: WorkbenchListWidget | undefined;
+
+	/**
+	 * Returns the last `IOpenEvent` that occured if known.
+	 */
+	readonly lastOpenContext: IOpenEvent<unknown> | undefined;
+
+	/**
+	 * Remember the last `IOpenEvent` that occured.
+	 */
+	setOpenContext(context: IOpenEvent<unknown>): IDisposable;
```

Comment:
Could we move something like this directly into the `IOpenerService`? That already knows `OpenOption`. With that the service could handle everything internally and the commands wouldn't be required to know this (tho the command do need some tweak so that they don't use the editor service directly).

Reply:
@jrieken isn't that risking the exposure of this "hack" to anyone that uses the opener service? What I like about this approach is that it is exclusively used for the 2 commands and nothing else and we know only extensions can call it.

Reply:
With that it is definitely widely available but I don't see it as a hack but as something really nice. Thinking of all the bugs I have fixed for "internal" features like outline, peek, go-to-links etc I would have really wanted this automagic behaviour a couple of years back. 

Tho, fine for me if you wanna keep it local (at least for now) but I think this approach is actually good and we should use it internally as well